[CoreML][ios_crash] Use special throw macro when encountering CoreML API errors

mcr229 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #86938

Error messages from TORCH_CHECK are stripped during production builds via  -DSTRIP_ERROR_MESSAGES. This diff introduces a new macro COREML_CHECK which will always preserve the error message. This macro is used when encountering errors produced by CoreML API calls so that we can heve enough context to debug.

Differential Revision: [D40351013](https://our.internmc.facebook.com/intern/diff/D40351013/)

mcr229 committed:
[CoreML][ios_crash] Use special throw macro when encountering CoreML API errors

Error messages from TORCH_CHECK are stripped during production builds via  -DSTRIP_ERROR_MESSAGES. This diff introduces a new macro COREML_CHECK which will always preserve the error message. This macro is used when encountering errors produced by CoreML API calls so that we can heve enough context to debug.

Differential Revision: [D40351013](https://our.internmc.facebook.com/intern/diff/D40351013/)

[ghstack-poisoned]

salilsdesai reviewed (approved):
![image](https://user-images.githubusercontent.com/25376640/195718626-f4c06f1a-c004-4f9d-9dd1-2904b501aa62.png)

mcr229 committed:
Update on "[CoreML][ios_crash] Use special throw macro when encountering CoreML API errors"


Error messages from TORCH_CHECK are stripped during production builds via  -DSTRIP_ERROR_MESSAGES. This diff introduces a new macro COREML_CHECK which will always preserve the error message. This macro is used when encountering errors produced by CoreML API calls so that we can heve enough context to debug.

Differential Revision: [D40351013](https://our.internmc.facebook.com/intern/diff/D40351013/)

[ghstack-poisoned]

mcr229 committed:
Update on "[CoreML][ios_crash] Use special throw macro when encountering CoreML API errors"


Error messages from TORCH_CHECK are stripped during production builds via  -DSTRIP_ERROR_MESSAGES. This diff introduces a new macro COREML_CHECK which will always preserve the error message. This macro is used when encountering errors produced by CoreML API calls so that we can heve enough context to debug.

Differential Revision: [D40351013](https://our.internmc.facebook.com/intern/diff/D40351013/)

[ghstack-poisoned]
